wrapper/src/main/java/software/amazon/jdbc/ConnectionPluginManager.java

```diff
@@ -118,7 +118,7 @@ public class ConnectionPluginManager implements CanReleaseResources, Wrapper {
   public ConnectionPluginManager(
       final @NonNull ConnectionProvider defaultConnProvider,
       final @Nullable ConnectionProvider effectiveConnProvider,
-      final @NonNull ConnectionWrapper connectionWrapper,
+      final @Nullable ConnectionWrapper connectionWrapper,
```

Comment:
I think it make sense to find all references of connectionWrapper and add null checks there. For debug build we can throw an exception, and for production build we just do nothing. Not a first priority.